src/Persistence/Mapping/AbstractClassMetadataFactory.php

```diff
@@ -205,17 +206,16 @@ public function getMetadataFor(string $className)
                     $this->wakeupReflection($cached, $this->getReflectionService());
                 } else {
                     $loadedMetadata = $this->loadMetadata($realClassName);
-                    $classNames     = array_combine(
-                        array_map([$this, 'getCacheKey'], $loadedMetadata),
-                        $loadedMetadata
-                    );
+                    $cacheItems     = $this->cache->getItems(array_map([$this, 'getCacheKey'], $loadedMetadata));
 
-                    foreach ($this->cache->getItems(array_keys($classNames)) as $item) {
-                        if (! isset($classNames[$item->getKey()])) {
-                            continue;
-                        }
+                    if ($cacheItems instanceof Traversable) {
+                        $cacheItems = iterator_to_array($cacheItems);
+                    }
+
+                    $loadedMetadataItems = array_combine($loadedMetadata, $cacheItems);
```

Comment:
this logic assumes that `$cacheItems is in the same order than `$loadedMetadataItems`. However, I don't think PSR-6 makes any guarantee about the order of items in the returned result.
The CacheItemPoolInterface phpdoc only says that the returned iterable is keyed by cache keys.

Reply:
You're right. This led me to an even simpler solution for this code part. Check my newer commit. It iterates over loaded metadata and get item for each, but I think it's not too frequent that $loadedMetadata's count is more than one, so it's maybe not a performance issue.